lib/src/mock-server/proxy.ts

```diff
@@ -35,6 +35,11 @@ export function proxyRequest({
     response.writeHead(res.statusCode ?? response.statusCode, res.headers);
     res.pipe(response);
   });
+  proxyRequest.on("error", e => {
+    console.error(`Failed to proxy request: ${e}`, e.stack);
+    response.statusCode = 500;
+    response.send();
+  });
```

Comment:
We previously were not handling any form of network or connection error on the proxied request. Any errors would cause the mock server to die. This instead logs the error and returns a 500 response.